crates/bevy_ecs/src/component.rs

```diff
@@ -657,14 +653,14 @@ impl Components {
         func: impl FnOnce() -> ComponentDescriptor,
     ) -> ComponentId {
         let components = &mut self.components;
-        let index = self.resource_indices.entry(type_id).or_insert_with(|| {
+        let component_id = self.resource_indices.entry(type_id).or_insert_with(|| {
             let descriptor = func();
-            let index = components.len();
-            components.push(ComponentInfo::new(ComponentId(index), descriptor));
-            index
+            let component_id = ComponentId(components.len());
+            components.push(ComponentInfo::new(component_id, descriptor));
+            component_id
         });
 
-        ComponentId(*index)
+        *component_id
```

Comment:
If I'm not mistaken, this can be returned directly:

```rust
        *self.resource_indices.entry(type_id).or_insert_with(|| {
            let descriptor = func();
            let component_id = ComponentId(components.len());
            components.push(ComponentInfo::new(component_id, descriptor));
            component_id
        })
```